rio_stac/stac.py

```diff
@@ -0,0 +1,205 @@
+) -> Dict:
+    """Get Raster Metadata."""
+    metadata: Dict[str, Any] = {}
+
+    metadata["name"] = src_dst.name
+
+    # To Do: handle non-geo data
+    bbox = warp.transform_bounds(
+        src_dst.crs, "epsg:4326", *src_dst.bounds, densify_pts=21
+    )
+
+    metadata["bbox"] = list(bbox)
+    metadata["footprint"] = bbox_to_geom(bbox)
+
+    # GDAL Metadata
+    metadata["tags"] = src_dst.tags()
+    metadata["width"] = src_dst.width
+    metadata["height"] = src_dst.height
+    metadata["dtype"] = src_dst.dtypes[0]
+
+    # Proj
+    try:
+        epsg = src_dst.crs.to_epsg() or None
+    except AttributeError:
+        epsg = None
+
+    metadata["proj"] = {
+        "epsg": epsg,
+        "geometry": bbox_to_geom(src_dst.bounds),
+        "bbox": list(src_dst.bounds),
+        "shape": [src_dst.height, src_dst.width],
+        "transform": list(src_dst.transform),
+    }
+
+    # rasterio can't reproduce wkt2
+    # ref: https://github.com/mapbox/rasterio/issues/2044
+    # if epsg is None:
+    #      metadata["proj"]["wkt2"] = src_dst.crs.to_wkt()
+
+    return metadata
+
+
+def get_media_type(
+    src_dst: Union[DatasetReader, DatasetWriter, WarpedVRT, MemoryFile],
+    input_media_type: Optional[pystac.MediaType] = None,
+) -> Optional[pystac.MediaType]:
+    """Define or validate MediaType."""
+    driver = src_dst.driver
+
+    output_media_type = input_media_type
+    if driver == "GTiff":
+        if src_dst.crs:
+            output_media_type = pystac.MediaType.GEOTIFF
+        else:
+            output_media_type = pystac.MediaType.TIFF
+
+    elif driver in [
+        "JP2ECW",
+        "JP2KAK",
+        "JP2LURA",
+        "JP2MrSID",
+        "JP2OpenJPEG",
+        "JPEG2000",
+    ]:
+        output_media_type = pystac.MediaType.JPEG2000
+
+    elif driver in ["HDF4", "HDF4Image"]:
+        output_media_type = pystac.MediaType.HDF
+
+    elif driver in ["HDF5", "HDF5Image"]:
+        output_media_type = pystac.MediaType.HDF5
+
+    elif driver == "JPEG":
+        output_media_type = pystac.MediaType.JPEG
+
+    elif driver == "PNG":
+        output_media_type = pystac.MediaType.PNG
+
+    if not output_media_type:
+        warnings.warn("Could not determine the media type from GDAL driver.")
+        return input_media_type
+
+    if input_media_type == pystac.MediaType.COG and driver != "GTiff":
+        raise Exception("Input file is not a GeoTIFF.")
+
+    elif input_media_type and output_media_type != input_media_type:
+        raise Exception("MediaType do not match.")
+
+    return output_media_type
+
+
+def create_stac_item(
+    source: Union[str, DatasetReader, DatasetWriter, WarpedVRT, MemoryFile],
+    input_datetime: Optional[datetime.datetime] = None,
+    extensions: Optional[List[str]] = None,
+    collection: Optional[str] = None,
+    item_properties: Optional[Dict] = None,
+    id: Optional[str] = None,
+    assets: Optional[Dict[str, pystac.Asset]] = None,
```

Comment:
@geospatial-jeff, I think this will be enough for your use case. I don't really want to add more complexity to the CLI so I think enabling the user to either pass a set of Assets in the API or use the source as the uniq asset is fine 🤷‍♂️

it's use `input assets` **OR** `source as asset` to avoid having twice the asset in the item. User has to understand that. 

Because we now returns the `pystac.item` from  `create_stac_item` it should be pretty straight forward to update it.